mango-python/bdgenomics/mango/variantViz.py

```diff
@@ -0,0 +1,442 @@
+                u'missense_variant',
+                u'stop_lost',
+                u'start_lost',
+                u'splice_donor_variant',
+                u'splice_acceptor_variant',
+                u'splice_region_variant',
+                u'frameshift_variant',
+                u'stop_gained'
+            ]
+
+        effectCounts = variantRDD.toDF().select("annotation").rdd.map(lambda w: w[0].asDict()['transcriptEffects']) \
+            .flatMap(lambda m: m).map(lambda g: g.asDict()['effects']).flatMap(lambda a: a).filter(lambda i: i in includeList).countByValue()
+
+
+        self.sc = ss.sparkContext
+        self.effectCounts = effectCounts
+
+    def plot(self, testMode = False):
+
+        if (not testMode):
+          plt.rcdefaults()
+          fig, ax = plt.subplots()
+          effects = self.effectCounts.keys()
+          y_pos = np.arange(len(effects))
+          counts = [ self.effectCounts[item] for item in effects]
+          ax.barh(y_pos, counts, align='center', color='green', ecolor='black')
+          ax.set_yticks(y_pos)
+          ax.set_yticklabels(effects)
+          ax.xaxis.set_label_text("Variant Count")
+          plt.title("Variant Count By Functional Consequence")
+          plt.show()
+
+        return dict(self.effectCounts)
+
+class VariantCountByGene(object):
+    """
+    Plot Histogram of variant count by Gene
+    """
+    def __init__(self,ss,variantRDD, so_term='missense_variant', bins=100):
+        """
+        :param ss: SparkContext 
+        :param variantRDD: bdgenomics VariantRDD
+        :param so_term: sequence ontology term to count within genes
+        :param bins: number of bins in histograms
+        """
+
+        def compute(dd1):
+          effectList = [ effect for transcript in dd1['transcriptEffects'] for effect in transcript['effects'] ]
+          geneList = [ transcript['geneId'] for transcript in dd1['transcriptEffects'] ]
+          if len(geneList)==0:
+            geneList = list("NoGene")
+          mostSigEffect = most_sig_effect(effectList)
+          return [mostSigEffect, geneList[0]]
+
+        y= variantRDD.toDF().rdd.map(lambda w: w.asDict()).map(lambda p: p['annotation']).filter(
+          lambda x: x['attributes']['AF_NFE'] != ".").map(
+          lambda d: ( compute(d)[0],compute(d)[1])).filter(
+          lambda f: f[0] == so_term and f[1] != 'NoGene').countByValue()
+
+        self.data = y.values()
+        self.bins = bins
+        self.so_term = so_term
+
+    def plot(self, testMode = False):
+
+        if (not testMode):
+          plt.hist(self.data, self.bins)
+          plt.title("Count of " + self.so_term + " per gene")
+          plt.xlabel("Number of Variants")
+          plt.ylabel("Number of Genes")
+          plt.show()
+        return self.data
+
+class VariantsGenomicRegion(object):
+    """
+    Plot variant density across chromosome region
+    """
+    def __init__(self,ss,variantRDD, start = 0, end = 1000000000, contigName="22"):
```

Comment:
why are these set? start = 0, end = 1000000000, contigName="22"

Reply:
I mean it as an example of approrpiate parameterws, but you are right, it really makes more sense to have no default, I will get rid of